config/v1/tests/ingresses.config.openshift.io/SetEIPForNLBIngressController.yaml

```diff
@@ -0,0 +1,224 @@
+apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this
+name: "Ingress"
+crdName: ingresses.config.openshift.io
+featureGate: SetEIPForNLBIngressController
+tests:
+  onCreate:
+    - name: Should not allow to set NLB parameters when LBType is Classic.
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Ingress
+        metadata:
+          name: cluster
+        spec:
+          loadBalancer:
+            platform:
+              type: AWS
+              aws:
+                type: Classic
+                networkLoadBalancer:
+                  eipAllocations:
+                  - eipalloc-12345
+                  - eipalloc-12346
+      expectedError: "Network load balancer parameters are allowed only when load balancer type is NLB."
+    - name: Should allow to set NLB parameters when LBType is NLB.
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Ingress
+        metadata:
+          name: cluster
+        spec:
+          loadBalancer:
+            platform:
+              type: AWS
+              aws:
+                type: NLB
+                networkLoadBalancer:
+                  eipAllocations:
+                  - eipalloc-1234567891011abcd
+                  - eipalloc-1234695789abcdefa
+      expected: |
+        apiVersion: config.openshift.io/v1
+        kind: Ingress
+        metadata:
+          name: cluster
+        spec:
+          loadBalancer:
+            platform:
+                type: AWS
+                aws:
+                  type: NLB
+                  networkLoadBalancer:
+                    eipAllocations:
+                    - eipalloc-1234567891011abcd
+                    - eipalloc-1234695789abcdefa
+    - name: Should not be able to create default ingresscontroller if the value of eipAllocations is not unique.
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Ingress
+        metadata:
+          name: cluster
+        spec:
+          loadBalancer:
+            platform:
+              type: AWS
+              aws:
+                type: NLB
+                networkLoadBalancer:
+                  eipAllocations:
+                  - eipalloc-12345
+                  - eipalloc-12345
+      expectedError: "eipAllocations: Invalid value: \"array\": eipAllocations cannot contain duplicates"
```

Comment:
The error message has `"array"`?  Is that expected?

Reply:
eipAllocations is an array. So it is expected from my understanding. Is there any other type we should be expecting ?